Update TRAVIS config: tests splitted

notoraptor opened the pull request:
theano/tensor tests are splitted into:
- all theano/tensor tests excluding test_basic
- just test_basic

`mode=FAST_COMPILE` tests are splitted into:
- all tests excluding theano/tensor tests
- just theano/tensor tests

@nouiz

nouiz commented on .travis.yml:
```diff
@@ -43,15 +43,20 @@ install:
-      env: PART="-e test_flake8.py ." THEANO_FLAGS="mode=FAST_COMPILE"
+      env:
+        - PART="-e theano/tensor test_flake8.py ." THEANO_FLAGS="mode=FAST_COMPILE"
+        - PART="theano/tensor" THEANO_FLAGS="mode=FAST_COMPILE"
```
This won't do what we want I think. I think it should be:

        - PART="-e theano/tensor -e test_flake8.py ." THEANO_FLAGS="mode=FAST_COMPILE"
        - PART="theano/tensor -e test_flake8.py" THEANO_FLAGS="mode=FAST_COMPILE"

I'm not sure we can pass many -e parameter to nosetests, double check that.

Also, line 46 give the full path to test_basic.py, while here we pass only the file name. Make sure that passing the full path is valid. You can check in git log of this file, we where doing that in the path.

notoraptor commented:
New update @nouiz !

notoraptor commented:
New correction. I find a travis validator here: http://lint.travis-ci.org/

Based on it, I have modified "matrix" section in the config file.

nouiz commented:
I killed the "push" travis run. When the "pr" run pass, we can merge.

nouiz commented:
In fact, I forgot, when the "pr" travis run pass, we need to check the split run output and check that we executed the right number of tests compared to the previous run in travis. Just to be sure we didn't loose some tests. I think the PR is good, but it is better to double check that.

notoraptor commented:
Hello !

As some runs yet take from 40 minutes to more than 1 hour, I suggest to split the tests differently. This is my suggestion:

original: `"theano/tensor"`
split:
- `"theano/tensor --exclude-dir=theano/tensor/nnet"`
- `"theano/tensor/nnet"`

PS: `test_basic.py` juste takes 4 to 6 minutes, so it does not make such difference, then I suggest not considerer it as a split point !

For `mode=FAST_COMPILE`:
original: `"-e test_flake8.py ."`
split:
- `"--exclude-dir=theano/tensor -e test_flake8.py ."`
- `"theano/tensor --exclude-dir=theano/tensor/nnet"`
- `"theano/tensor/nnet"`

(we isolate `theano/tensor` and then we split it the same manner as above)

Also, about comparing number of tests to previous runs, it seemes that TRAVIS runs on a clone of the master branch of the official Theano repository, on which it applies the tested PR. At the top of any run log ( e.g. https://travis-ci.org/Theano/Theano/jobs/176130787 ), we have the following commands:

```
$ git clone --depth=50 https://github.com/Theano/Theano.git Theano/Theano
$ cd Theano/Theano
$ git fetch origin +refs/pull/5226/merge:
$ git checkout -qf FETCH_HEAD
```

So, to compare correctly the number of tests executed, for every travis job (i.e. every value of "PART"), it seems better to:
1. get the time the travis job started,
2. find the latest commit to theano/master that was merged just before the travis job started,
3. use this commit as reference.

But taking account that some commits do not have TRAVIS runs !



@nouiz
